### ISSUE TITLE
Filter function in `mutate` returns `true`, but does not seem to invalidate cache (using `useSWRInfinite`)

### ISSUE THREAD

**Issue by s1gr1d** [vercel/swr#2390]
# Bug report

## Description / Observed Behavior

I am using `useSWRInfinite` with `getKey`, which returns an array. When using the global `mutate` from `useSWRConfig`, I'm passing a function to perform a multiple mutate. This filter function checks if the key is an array and if the searched string is included.

However, the filter function returns `true` for the key that should be re-validated, but the API is not requested to get new data. So I'm having the issue that the UI is not updating, unless I reload the page.

I created a Code Sandbox: https://codesandbox.io/s/multiple-mutate-infiniteswr-okty3z

## Expected Behavior

When updating multiple keys from `useSWR`, it works perfect (look at the Code Sandbox example), but when using `useSWRInfinite` it does not work. 

I expect that the API function is called again to request new data after the mutate filter function returns `true` for the cache entry.

## Repro Steps / Code Example

I have this issue in a bigger project, which I cannot post here. But I provided a small CodeSandbox, where the same issue occurs.

[CodeSandbox Link](https://codesandbox.io/s/multiple-mutate-infiniteswr-okty3z) 

## Additional Context

SWR version: 2.0.2 (basically every version after 2.0.0)

**Comment by koba04** [comment]
@siggy17
Yes, [Mutate multiple item](https://swr.vercel.app/docs/mutation#mutate-multiple-items) doesn't support `useSWRInfinite`, so you have to use `mutate` returned from `useSWRInfintie` to revalidate it. But the `mutate` doesn't accept a filter function.

We should document it and I'll consider a way to support it with `useSWRInfinite`. Thank you for your feedback!

**Comment by s1gr1d** [comment]
@koba04 Thank you for your reply! Yes, documenting that would be absolutely useful. And also supporting "multiple mutate" with `useSWRInfinite` would be nice. The reason I stumbled upon this bug was that there was no built-in functionality for `multiple mutate` before, and we had our own hook for it (see below). After upgrading to `2.0.0` I replaced this custom hook with the filter function "multiple mutate" and then the errors occurred. 

<details>
<summary>Custom hook: useMultipleMutate</summary>
<br>

```
import { useSWRConfig } from 'swr';
import { useCallback } from 'react';

export const useMultipleMutate = () => {
  const { cache, mutate } = useSWRConfig();
  const multipleMutate = useCallback(
    (cacheKey: string) => {
      const matcher = new RegExp(`^${cacheKey}`);
      if (!(cache instanceof Map)) {
        throw new Error('matchMutate requires the cache provider to be a Map instance');
      }

      const keys = [];

      for (const key of cache.keys()) {
        if (matcher.test(key)) {
          keys.push(key);
        }
      }

      const mutations = keys.map((key) => mutate(key));
      return Promise.all(mutations);
    },
    [cache, mutate],
  );

  return { multipleMutate };
};
```
</details>
